homework/index.js

```diff
@@ -30,17 +32,134 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  function renderError(error) {
+    const container = document.getElementById('container');
+    container.innerHTML = '';
+    createAndAppend('div', container, {
+      text: error.message,
+      class: 'error',
+    });
+  }
+
+  function showRepositoryInfo(repository) {
+    const leftHand = document.getElementById('left-hand');
+    leftHand.innerHTML = '';
+    const tableElement = createAndAppend('table', leftHand);
+    const rowForName = createAndAppend('tr', tableElement);
+    createAndAppend('th', rowForName, {
+      text: 'Repository:',
+    });
+    const tdForName = createAndAppend('td', rowForName);
+    createAndAppend('a', tdForName, {
+      href: repository.html_url,
+      target: '_blank',
+      text: repository.name,
+    });
+    const rowForDescription = createAndAppend('tr', tableElement);
+    createAndAppend('th', rowForDescription, {
+      text: 'Description:',
+    });
+    createAndAppend('td', rowForDescription, { text: repository.description });
+    const rowForForks = createAndAppend('tr', tableElement);
+    createAndAppend('th', rowForForks, {
+      text: 'Forks:',
+    });
+    createAndAppend('td', rowForForks, { text: repository.forks_count });
+    const rowForUpdated = createAndAppend('tr', tableElement);
+    createAndAppend('th', rowForUpdated, {
+      text: 'Updated:',
+    });
+    createAndAppend('td', rowForUpdated, {
+      text: new Date(repository.updated_at).toLocaleString('en-GB'),
+    });
+  }
+
+  function setRepository(repository) {
+    const rightHand = document.getElementById('right-hand');
+    rightHand.innerHTML = '';
+    fetchJSON(repository.contributors_url, (error, users) => {
+      if (error) {
+        renderError(error);
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+      const h3Element = createAndAppend('h3', rightHand, {
+        text: 'Contributors:',
+      });
+      if (users.length === 0) {
+        h3Element.textContent = 'Contributor Not Found!';
+        return;
+      }
+      const ulElement = createAndAppend('ul', rightHand, {
+        class: 'contributors',
+      });
+      users.forEach(elem => {
+        const liElement = createAndAppend('li', ulElement, {
+          class: 'contributors',
+        });
+        createAndAppend('img', liElement, {
+          class: 'user-image',
+          src: elem.avatar_url,
+        });
+        createAndAppend('span', liElement, {
+          class: 'login',
+          text: elem.login,
+        });
+        createAndAppend('span', liElement, {
+          class: 'counter',
+          text: elem.contributions,
+        });
+        liElement.onclick = () => {
```

Comment:
Responding to the click event works fine for users with normal vision, but visually impaired users need to able to navigate solely with the keyboard. They would need to be able to press Enter and go to the contributor's page while the list item has the focus. The best way to achieve is to use a regular `a` tag that wraps the entire inner content of the `li` element and has an `href` with the correct url. The default behaviour of the `a` tag meets all the requirements for ARIA compliance (part of the tab order, clickable and can be activated with the Enter key).